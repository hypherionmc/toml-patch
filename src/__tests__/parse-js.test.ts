import parseJS from '../parse-js';
import toTOML from '../to-toml';

const value = {
  a: '1',
  b: 2,
  c: 3.14,
  d: true,
  e: new Date('1979-05-27T07:32:00Z'),
  f: {
    g: ['h', 'i', 'j'],
    k: [
      { l: 'm' },
      {
        n: 'o',
        p: {
          toJSON() {
            return 'qrs';
          }
        }
      }
    ]
  }
};

test('it should parse object', () => {
  expect(parseJS(value)).toMatchSnapshot();
});

test.only('it should be properly formatted', () => {
  expect(toTOML(parseJS(value))).toMatchSnapshot();
  expect(toTOML(parseJS(value, { trailingComma: true }))).toMatchSnapshot();
});
