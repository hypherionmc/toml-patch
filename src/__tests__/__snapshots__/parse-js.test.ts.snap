// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should be properly formatted 1`] = `
"a = \\"1\\"
b = 2
c = 3.14
d = true
e = 1979-05-27T07:32:00.000Z

[f]
g = [ \\"h\\", \\"i\\", \\"j\\" ]
k = [ { l = \\"m\\" }, { n = \\"o\\", p = \\"qrs\\" } ]

[[t]]
u = \\"v\\"

[[t]]
w = \\"x\\"

[[t]]
y = \\"z\\"
"
`;

exports[`it should be properly formatted 2`] = `
"a = \\"1\\"
b = 2
c = 3.14
d = true
e = 1979-05-27T07:32:00.000Z

[f]
g = [\\"h\\", \\"i\\", \\"j\\",]
k = [{l = \\"m\\",}, {n = \\"o\\", p = \\"qrs\\",},]

[[t]]
u = \\"v\\"

[[t]]
w = \\"x\\"

[[t]]
y = \\"z\\"
"
`;
